os/board/rtl8721csm/src/component/common/bluetooth/realtek/sdk/example/ble_central/ble_tizenrt_client.c

```diff
@@ -139,9 +312,22 @@ trble_result_e rtw_ble_client_start_scan_with_filter(trble_scan_filter* scan_par
     } else {
         debug_print("\r\n[%s] set scan info success", __FUNCTION__);
     }
-    
-    if(ble_tizenrt_client_send_msg(BLE_TIZENRT_START_SCAN, NULL) == false)
+
```

Comment:
Can you change the usage of 'le_scan_info_filter' like below?
```
if (scan_parm->raw_data_length != 0) {
        if(!le_scan_info_filter(true, 0, scan_parm->raw_data_length, scan_parm->raw_data))
        {
            printf("\r\n[%s] set scan info fail !!", __FUNCTION__);
            return TRBLE_FAIL;
        } else {
            debug_print("\r\n[%s] set scan info success", __FUNCTION__);
        }
    } else {
        if(le_scan_info_filter(false, 0, 0, NULL))
        {
            debug_print("\r\n[%s] disable scan info filter success", __FUNCTION__);
        } else {
            debug_print("\r\n[%s] disable scan info filter fail!!!", __FUNCTION__);
        }
    }
}
```

I want that the filters work independently.
For example, someone needs to filter with only timers.

Reply:
@ziyik 
Yes, you are right.
When 'scan_parm->raw_data_length' is 0, clear raw data filter.